src/test/allocator_tests.cpp

```diff
@@ -233,4 +235,41 @@ BOOST_AUTO_TEST_CASE(lockedpool_tests_live)
     BOOST_CHECK(pool.stats().used == initial.used);
 }
 
+BOOST_AUTO_TEST_CASE(accounting_allocation_test)
+{
+    size_t total = 0;
+    size_t dummy = 0;
+
+    {
+        typedef std::unordered_set<int, std::hash<int>, std::equal_to<int>, memusage::AccountingAllocator<int>> container_type;
+        container_type container1({}, 2, std::hash<int>(), std::equal_to<int>(), memusage::AccountingAllocator<int>(dummy)); // container1 is accounted for in 'dummy'
+        container1.insert(6);
+        {
+            container_type container2(1, std::hash<int>(), std::equal_to<int>(), memusage::AccountingAllocator<int>(total)); // container2 is accounted for in 'total'
+            container2.insert(5);
+            container2.insert(2);
+            container2.insert(3);
+            container2.erase(2);
+            BOOST_CHECK(total > 0);
+            size_t old = total;
+            container1 = std::move(container2); // container1 is now accounted for in 'total'
+            BOOST_CHECK(total >= old);
```

Comment:
65fc229fabbc7889baca9c50ee35507543a6682e
```suggestion
            BOOST_CHECK(total == old);
```